make.py

```diff
@@ -0,0 +1,252 @@
+install_parser.set_defaults(command='install')
+install_parser.add_argument('prefix', type=str,
+                            help='Where to install Langkit dependencies')
+install_parser.add_argument('--gnat-install-dir', type=str, default=None,
+                            help='Specify where GNAT is installed. If not'
+                                 'provided, assume it is installed where '
+                                 'Langkit must be installed.')
+install_parser.add_argument('--gnatcoll-bindings-install-dir', type=str,
+                            default=None,
+                            help='Specify where GNATCOLL bindings are '
+                                 'installed. If not provided, assume it is '
+                                 'installed where Langkit must be installed.')
+install_parser.add_argument('--libgpr-install-dir', type=str,
+                            default=None,
+                            help='Specify where libgpr is installed. If not '
+                                 'provided, assume it is installed where '
+                                 'Langkit must be installed.')
+
+
+ROOT_DIR = os.path.dirname(__file__)
+
+if ROOT_DIR != '':
+    os.chdir(ROOT_DIR)
+
+LIBPYTHONLANG_DIR = os.path.join('contrib', 'python')
+
+
+@contextmanager
+def in_dir(path):
+    """
+    Move the working directly to the given path for the span of this context
+    manager.
+
+    :type path: str
+    """
+    old_dir = os.getcwd()
+    os.chdir(path)
+    yield
+    os.chdir(old_dir)
+
+
+def pip_install(setup_dir):
+    """
+    Install the python package located in the given directory.
+
+    :type setup_dir: str
+    :raise CalledProcessError if installation terminates with non-zero status
+    """
+    with in_dir(setup_dir):
+        subprocess.check_call([sys.executable, '-m', 'pip', 'install', '.'])
+
+
+def copy_all_shared_libs(from_dir, destination, *patterns):
+    """
+    Copy all shared libraries from anywhere in a given source tree to a given
+    destination folder.
+
+    Only the libraries that satisfy the given simple patterns are copied.
+    Given patterns must NOT be regular expressions, but simple strings.
+    A library will match if the pattern is a substring of the library's name.
+
+    :type from_dir: str
+    :type destination: str
+    :type patterns: *str
+    """
+    for root, _, filenames in os.walk(from_dir):
+        for filename in filenames:
+            if any(n in filename for n in patterns):
+                if any(ext in filename for ext in (".so", ".dll", ".dylib")):
+                    shutil.copy(os.path.join(root, filename), destination)
+
+
+def manage(lang_dir, verb, build_mode, *add_args, **kwargs):
+    """
+    Run the manage.py script from of a Langkit-based language definition.
+
+    :param str verb: 'generate', 'build', 'install', etc.
+    :param str build_mode: 'dev' or 'prod'.
+    :param str|None rpath: if set, defines the RPATH to include in built
```

Comment:
If `rpath` is documented, why not moving it out of `kwargs`? And why have `kwargs` at all?

Reply:
I wanted `rpath` to be an optional argument, but also allow additional arguments to pass to `manage.py` through `*args`. I'm not sure how to achieve that any other way... Or do you prefer I remove the `*args` and ask for a list explicitly?

Reply:
I see… thanks. Actually given that we pass `-R` nightly, I would suggest to remove this altoghether and always build with `--gargs=-R`.

Reply:
Can it not be useful for a development build though to keep the rpaths? (For example if you want to `build` and `setenv` to develop on langkit)

Reply:
Not really: for the Python bindings, running the install part, which bundles all libs thus removing the need to fiddle with RPATHs, is very cheap, and for Ada programs, these can be built with RPATHs (happens by default), so the RPATHs in libraries are useless. So I really think we could remove it here :-)